.github/workflows/main.yaml

```diff
@@ -49,16 +69,46 @@ jobs:
         with:
           name: release-artifacts
           path: release-artifacts
+  build-windows:
+    name: build windows
+    runs-on: ubuntu-latest
+    container: golang:1.16.1-alpine3.12
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v1
+        with:
+          fetch-depth: 1
+      - name: Build
+        run: |
+          scripts/alpine-setup.sh
+          GOOS=windows GOARCH=amd64 make all
+        shell: sh
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - name: Archive release artifacts
+        uses: actions/upload-artifact@v1
+        with:
+          name: release-artifacts-windows
+          path: release-artifacts
   release:
     name: Release
-    needs: build
+    needs:
+      [test, pre-commit, build, build-darwin, build-windows]
     runs-on: ubuntu-latest
     if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags/')
     steps:
       - uses: actions/download-artifact@v1
         with:
           name: release-artifacts
           path: release-artifacts
+      - uses: actions/download-artifact@v1
+        with:
+          name: release-artifacts-darwin
+          path: release-artifacts
```

Comment:
I think this will put the artifacts in the `release-artifacts` directory, i.e. the next upload release asset step will fail.